libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/entitlements/FilesEntitlement.java

```diff
@@ -298,15 +332,17 @@ public static FilesEntitlement build(List<Object> paths) {
         }
         List<FileData> filesData = new ArrayList<>();
         for (Object object : paths) {
-            Map<String, String> file = new HashMap<>((Map<String, String>) object);
-            String pathAsString = file.remove("path");
-            String relativePathAsString = file.remove("relative_path");
-            String relativeTo = file.remove("relative_to");
-            String pathSetting = file.remove("path_setting");
-            String relativePathSetting = file.remove("relative_path_setting");
-            String modeAsString = file.remove("mode");
-            String platformAsString = file.remove("platform");
-            String ignoreUrlAsString = file.remove("ignore_url");
+            Map<String, Object> file = new HashMap<>((Map<String, String>) object);
+            String pathAsString = (String) file.remove("path");
+            String relativePathAsString = (String) file.remove("relative_path");
+            String relativeTo = (String) file.remove("relative_to");
+            String pathSetting = (String) file.remove("path_setting");
+            String relativePathSetting = (String) file.remove("relative_path_setting");
+            String modeAsString = (String) file.remove("mode");
+            String platformAsString = (String) file.remove("platform");
+            String ignoreUrlAsString = (String) file.remove("ignore_url");
+            Boolean exclusiveBoolean = (Boolean) file.remove("exclusive");
```

Comment:
Can we have a better error message in case this is a string?

Reply:
Fixed.